src/workloads/scale/ReadMemoryStressUntilFailure.yml

```diff
@@ -0,0 +1,131 @@
+SchemaVersion: 2018-07-01
+Owner: Service Architecture
+Description: |
+  Used to test performance of the server under heavy memory pressure caused by read
+  operations. This workload is expected to fail due to host(s) being unreachable as a
+  result of mongod(s) running out of memory.
+
+  To achieve this, documents are inserted in the form of {a: <id>, b: <random number>,
+  c: <fill 520 KB>}. Many threads are then spawned to run aggregate sort on a number of
+  documents. The document size and the number of documents to sort by each thread are
+  tuned to be just under 100 MB, which is the memory limit for operations before
+  spilling to disk. The threads would therefore each cause a close-to-max amount of
+  memory to be used. Increasing the number of threads should cause the host(s) that
+  process the operations to fail due to out-of-memory errors.
+
+Keywords:
+- scale
+- memory stress
+- aggregate
+- sort
+- insert
+- fail
+- oom
+- out of memory
+
+GlobalDefaults:
+  dbname: &DBName memorystress
+  MaxPhases: &MaxPhases 20
+
+  LoadThreads: &LoadThreads 128
+  LoadBatchSize: &LoadBatchSize 500
+
+  NumberOfDocuments: &NumDocs 20096
```

Comment:
Is this number significant in any way other than being some "large" number? If not, what's the criteria that we want to document for what this number should be?

Reply:
`NumberOfDocuments` is just a multiple of `NumberOfDocumentsPerWorker`. This number is large enough that `NumberOfDocumentsPerWorker` and `DocumentSize` contribute to an almost 100MB document. These numbers are tunable, but if one number is tuned, then the others need to be as well in order to match the constraints (of 100MB).

So it is just a large number, and it can be changed. `DocumentSize` and `NumberOfDocumentsPerWorker` are both arbitrary but reasonable numbers.